src/main/java/org/folio/inventory/dataimport/handlers/matching/MarcBibliographicMatchEventHandler.java

```diff
@@ -0,0 +1,126 @@
+import io.vertx.core.json.Json;
+import org.folio.DataImportEventPayload;
+import org.folio.HoldingsRecord;
+import org.folio.inventory.common.Context;
+import org.folio.inventory.common.api.request.PagingParameters;
+import org.folio.inventory.consortium.services.ConsortiumService;
+import org.folio.inventory.dataimport.handlers.matching.util.EventHandlingUtil;
+import org.folio.inventory.dataimport.util.ParsedRecordUtil;
+import org.folio.inventory.dataimport.util.ParsedRecordUtil.AdditionalSubfields;
+import org.folio.inventory.domain.HoldingsRecordCollection;
+import org.folio.inventory.domain.instances.InstanceCollection;
+import org.folio.inventory.storage.Storage;
+import org.folio.rest.jaxrs.model.Record;
+import org.folio.rest.jaxrs.model.RecordMatchingDto;
+
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.folio.ActionProfile.FolioRecord.HOLDINGS;
+import static org.folio.ActionProfile.FolioRecord.INSTANCE;
+import static org.folio.ActionProfile.FolioRecord.MARC_BIBLIOGRAPHIC;
+import static org.folio.DataImportEventTypes.DI_SRS_MARC_BIB_RECORD_MATCHED;
+import static org.folio.DataImportEventTypes.DI_SRS_MARC_BIB_RECORD_NOT_MATCHED;
+import static org.folio.inventory.dataimport.handlers.matching.util.EventHandlingUtil.getTenant;
+
+public class MarcBibliographicMatchEventHandler extends AbstractMarcMatchEventHandler {
+
+  private static final String HOLDINGS_LOADING_ERROR_MSG = "Failed to load holdings by instanceId: '%s' for matched MARC-BIB, jobExecutionId: '%s'";
+  private final Storage storage;
+
+  public MarcBibliographicMatchEventHandler(ConsortiumService consortiumService, HttpClient httpClient, Storage storage) {
+    super(consortiumService, DI_SRS_MARC_BIB_RECORD_MATCHED, DI_SRS_MARC_BIB_RECORD_NOT_MATCHED, httpClient);
+    this.storage = storage;
+  }
+
+  @Override
+  protected RecordMatchingDto.RecordType getMatchedRecordType() {
+    return RecordMatchingDto.RecordType.MARC_BIB;
+  }
+
+  @Override
+  protected String getMarcType() {
+    return MARC_BIBLIOGRAPHIC.value();
+  }
+
+  @Override
+  protected boolean isMatchingOnCentralTenantRequired() {
+    return true;
+  }
+
+  @Override
+  protected Future<List<Record>> postProcessMatchingResult(List<Record> records, DataImportEventPayload eventPayload) {
+    if (records.size() == 1) {
+      Record matchedRecord = records.get(0);
+      String instanceId = ParsedRecordUtil.getAdditionalSubfieldValue(matchedRecord.getParsedRecord(), AdditionalSubfields.I);
+      String matchedRecordTenantId = getTenant(eventPayload);
+      Context context = EventHandlingUtil.constructContext(matchedRecordTenantId, eventPayload.getToken(), eventPayload.getOkapiUrl());
+      InstanceCollection instanceCollection = storage.getInstanceCollection(context);
+
+      if (isBlank(instanceId)) {
+        return Future.succeededFuture(records);
```

Comment:
@RuslanLavrov  Maybe add some logging in this case? When 999ff$i field will be empty.

Reply:
added logging